### ISSUE TITLE
List.ElementsAs crashes

### ISSUE THREAD

**Issue by sebhoss** [hashicorp/terraform-plugin-framework#528]
### Module version

```
github.com/hashicorp/terraform-plugin-framework v0.15.0
```

### Relevant provider source code

```go
import (
	"context"
	"github.com/hashicorp/terraform-plugin-framework/attr"
	"github.com/hashicorp/terraform-plugin-framework/types"
	"github.com/stretchr/testify/assert"
	"testing"
)

func TestListValue(t *testing.T) {
	values := []attr.Value{types.StringValue("refs/heads/main")}
	value, diags := types.ListValue(types.StringType, values)
	var refSpecs []string
	diags.Append(value.ElementsAs(context.TODO(), &refSpecs, false)...)
	assert.False(t, diags.HasError())
}
```

### Terraform Configuration Files

Not related to Terraform configuration

### Debug Output

https://gist.github.com/sebhoss/9c6f99a1a6014f2807c0b6589dc61de8

### Expected Behavior

The above code should do a roundtrip from Go `string` to `attr.Value` and back to `string`.

### Actual Behavior

Framework panics as seen in the log output. I can fix that on my end by switching from `types.ListValue` to `types.ListValueFrom` like this:

```go
func TestListValue(t *testing.T) {
	values := []attr.Value{types.StringValue("refs/heads/main")}
	value, diags := types.ListValueFrom(context.TODO(), types.StringType, values)
	var refSpecs []string
	diags.Append(value.ElementsAs(context.TODO(), &refSpecs, false)...)
	assert.False(t, diags.HasError())
}
```

### Steps to Reproduce

1. Copy the above code snippet into project
2. Run test

### References

I have not found any related tickets.

**Comment by bflad** [comment]
Hi @sebhoss 👋 Thank you for reporting this and sorry you ran into trouble here.

I believe this should already be fixed on `main` per #523 and will release with version 0.16.0 (along with those breaking changes).

**Comment by sebhoss** [comment]
Yep you are right - I tested against current `main` and the test passes just fine. Thanks for the quick reply!
